dags/sources/tasks/business_logic/db_write_type_action_suggestions.py

```diff
@@ -141,84 +180,127 @@ def insert_suggestion(
+            revision_acteur = RevisionActeur.objects.filter(
+                identifiant_unique=identifiant_unique
+            ).first()
+        parent = (
+            revision_acteur.parent
+            if revision_acteur and revision_acteur.parent
+            else None
+        )
+        suggestion_groupe = SuggestionGroupe(
+            suggestion_cohorte=suggestion_cohorte,
+            statut=constants.SUGGESTION_AVALIDER,
+            acteur=acteur,
+            revision_acteur=parent or revision_acteur,
+            contexte=row["contexte"],
+        )
+        suggestion_groupe.save()
+        suggestion = json.loads(row["suggestion"])
+        try:
+            contexte = json.loads(row["contexte"])
+        except TypeError:
+            contexte = {}
+
+        grouped_keys = [["latitude", "longitude"]]
+        flattened_grouped_keys = [item for sublist in grouped_keys for item in sublist]
+        keys_and_grouped_keys = [
+            [key]
+            for key in suggestion.keys()
+            if key not in ["location", "sous_categorie_codes", "action_codes"]
+            and key not in flattened_grouped_keys
+        ]
+        for grouped_key in grouped_keys:
+            if all(key in suggestion for key in grouped_key):
+                keys_and_grouped_keys.append(grouped_key)
+
+        # for key, value in suggestion.items():
+        for keys in keys_and_grouped_keys:
+            should_add_suggestion_unitaire = False
+            for key in keys:
+                if key not in contexte or contexte[key] != suggestion[key]:
+                    should_add_suggestion_unitaire = True
+                    break
+            if not should_add_suggestion_unitaire:
+                continue
+            values = []
+            for key in keys:
+                values.append(suggestion[key])
+            SuggestionUnitaire(
+                suggestion_groupe=suggestion_groupe,
+                statut=constants.SUGGESTION_AVALIDER,
+                acteur=acteur,
+                # Acteur will be updated when a source is updated
+                suggestion_modele="Acteur",
+                # fields should be grouped, ex : latitude and longitude
+                champs=keys,
+                valeurs=values,
+            ).save()
+            # Special case for SUPPRESSION which should be applied to its revision
+            # if it exists
+            if (
+                suggestion_cohorte.type_action == constants.SUGGESTION_SOURCE_SUPRESSION
+                and keys == ["statut"]
+                and revision_acteur
+            ):
+                SuggestionUnitaire(
+                    suggestion_groupe=suggestion_groupe,
+                    statut=constants.SUGGESTION_AVALIDER,
+                    acteur=acteur,
+                    revision_acteur=revision_acteur,
+                    suggestion_modele="RevisionActeur",
+                    champs=keys,
+                    valeurs=values,
+                ).save()
```

Comment:
question : a-t-on un enjeu de performance ici ? Je vois deux save, dans certaines situations ça pourrait impacter les perfs si on save deux fois pour chaque itération de la boucle non ?